apps/src/lib/node/ledger/events/log.rs

```diff
@@ -67,13 +67,12 @@ impl EventLog {
 
     /// Returns a new iterator over this [`EventLog`].
     #[inline]
-    pub fn iter_with_matcher<'query, 'log>(
-        &'log self,
-        matcher: dumb_queries::QueryMatcher<'query>,
-    ) -> impl Iterator<Item = &'log Event> + 'query
+    pub fn iter_with_matcher<'a, T>(
+        &'a self,
+        matcher: dumb_queries::QueryMatcher<'a, T>,
+    ) -> impl Iterator<Item = &Event> + 'a
     where
-        // the log should outlive the query
-        'log: 'query,
```

Comment:
we do get rid of this bit of lifetime gymnastics though, which is nice to the reader